Automattic/jetpack
# Add/assistant recommendation tracks

Update recommendation tracking to tell how often users click on the `Recommendations` tab of the dashboard when the "new recommendations" bubble is visible

#### Changes proposed in this Pull Request:

This PR adds a tracked event that triggers if the "new recommendations" bubble is visible on the dashboard next to the `Recommendations` tab. It also updates the tracks event that is recorded when the `Recommendations` tab is clicked to record whether or not the bubble was visible when it was clicked.

#### Other information:


#### Jetpack product discussion

P2: pbNhbs-3nL-p2#comment-10014

#### Does this pull request change what data or activity we track or use?

Yes, this PR adds a track event for when a user views the dashboard and the "new recommendations" notification bubble is visible.

It also adds extra data to the existing track click event when the `Recommendations` tab is clicked to determine if the bubble was visible when it was clicked.

#### Testing instructions:

1. Checkout this branch via `git switch add/assistant-recommendation-tracks`

2. Make sure you have the [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) chrome extension talked about in this fg article: PCYsg-cae-p2 (or at least have some way to monitor tracks)

3. Head to the Jetpack Dashboard and reset your Options via the `Reset Options (dev only)` item in the footer

![Screen Shot 2022-08-19 at 1 14 46 PM](https://user-images.githubusercontent.com/65001528/185691028-2ef6c620-e351-4b64-90be-afa03039e8eb.jpg)

4. Go install and activate any plugin from the plugin store to trigger the `security-plan` recommendation (or, if this is tested after [this PR](https://github.com/Automattic/jetpack/pull/25391) is merged, the `protect` recommendation)

5. Publish a new Post to trigger the `publicize` recommendation

6. Head back to the Jetpack Dashboard and confirm you see the Recommendations bubble with a "2" inside

![image](https://user-images.githubusercontent.com/65001528/185692169-74b29066-44cb-4f4d-a21b-2719592ab951.png)

7. Confirm in Tracks Vigilante (or however you monitor tracks) that a new track called `jetpack_recommendations_new_recommendation_bubble_visible` that looks like the image below was recorded. It should also records the `pathname` variable.

![image](https://user-images.githubusercontent.com/65001528/186223847-c0079449-b4de-4d68-b564-518d5add3414.png)

8. Click on the `Recommendations` tab to view the first recommendation

![image](https://user-images.githubusercontent.com/65001528/185692760-60df7683-b2b6-4bb8-8dcd-ef72c98acdc0.png)

9. Check your tracks again and confirm that a click event was recorded. It should indicate that the bubble was visible. Also confirm that `jetpack_recommendations_new_recommendation_bubble_visible` was not recorded this time.

![image](https://user-images.githubusercontent.com/65001528/186223966-59a38e47-d393-4736-9d8c-687ae45e7f80.png)

10. Go back to the `At a Glance` section and refresh the page. You should see another `jetpack_recommendations_new_recommendation_bubble_visible` track event recorded.

![image](https://user-images.githubusercontent.com/65001528/186224176-f34bec9c-5aad-4a74-bdae-8353a2b7b3be.png)

11. Click back on the recommendations tab and confirm there is another click event recorded that shows the bubble was visible.

![image](https://user-images.githubusercontent.com/65001528/186224333-b7868f99-841a-494b-9f3a-53ef992c90d5.png)

12. Go back to `At a Glance` and refresh the page. Confirm that the `jetpack_recommendations_new_recommendation_bubble_visible` event was absent this time

13. Click on the `Recommendations` tab again and confirm that a click event was recorded that indicates the bubble was _not_ visible.

![image](https://user-images.githubusercontent.com/65001528/186224469-c69780b9-9706-4dd2-a7e6-c781824f05b9.png)
